rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -163,6 +174,26 @@ void SequentialWriter::close()
   storage_factory_.reset();
 }
 
+void SequentialWriter::take_snapshot()
+{
+  circular_message_cache_->swap_buffers();
+  auto msg_list = circular_message_cache_->consumer_buffer()->data();
+
+  std::vector<std::shared_ptr<const rosbag2_storage::SerializedBagMessage>>
+  msg_vector(msg_list.size());
+
+  // Copy list into a vector needed by the storage writer
+  std::copy(msg_list.begin(), msg_list.end(), msg_vector.begin());
```

Comment:
I commented on this in the CircularBuffer PR, I think we should just be using a vector, the linkedlist is not giving us value, and this copy isn't necessarily cheap.